backend/src/apiserver/server/pipeline_server_test.go

```diff
@@ -47,6 +50,91 @@ func TestCreatePipeline_YAML(t *testing.T) {
 	assert.Equal(t, "pipeline description", newPipeline.Description)
 }
 
+func TestCreatePipeline_MultiUser(t *testing.T) {
```

Comment:
this might be a big ask, so it's OK if you don't do it in this PR.
It'll be very beneficial to merge these tests into one table driven test, because the test setup is just too long that we can hardly grasp what the test is about, besides from reading the test title.

Reply:
I can create a separate issue to track to refactor this is possibly other existing tests into table driven tests? what do you think?